dart/math/MathTypes.h

```diff
@@ -83,8 +85,28 @@ typedef Matrix<double, 6, 6> Matrix6d;
 typedef std::vector<Eigen::Vector3d> EIGEN_V_VEC3D;
 typedef std::vector<std::vector<Eigen::Vector3d > > EIGEN_VV_VEC3D;
 
+template <typename _Tp>
+using aligned_vector = std::vector<_Tp, Eigen::aligned_allocator<_Tp>>;
+
+template <typename _Key, typename _Tp, typename _Compare = std::less<_Key>>
+using aligned_map = std::map<_Key, _Tp, _Compare,
+    Eigen::aligned_allocator<std::pair<const _Key, _Tp>>>;
+
+template<typename _Tp, typename... _Args>
+inline std::shared_ptr<_Tp> make_aligned_shared()
```

Comment:
Oh, I see. The implicitly deleted copy constructor belongs to the templated class inside of ``<Eigen/.../util/Memory.h>``, not to WeldJoint::Properties. It seems really strange that this is happening specifically for WeldJoint::Properties. I'm thinking this must be an internal compiler error.

What happens when we make the no-argument version of ``Eigen::make_aligned_shared()`` use ``std::allocate_shared<_Tp>( Eigen::aligned_allocator<_Tp>() )``? Does the same error happen? I'm kind of concerned that using regular ``make_shared`` defeats the point of having the ``Eigen::make_aligned_shared()`` function.